sw/device/lib/base/memory_perftest.c

```diff
@@ -0,0 +1,216 @@
+}
+
+OTTF_DEFINE_TEST_CONFIG();
+
+// Each value of `expected_num_cycles` was determined experimentally by
+// testing on a CW310 FPGA with the following command:
+//
+//   $ ./bazelisk.sh test --copt -O2 --test_output=all \
+//       //sw/device/lib/base:memory_perftest_fpga_cw310
+//
+// There are a handful of reasons why the expected number of cycles for this
+// test might be inaccurate. Here are a few of them:
+//
+//   (1) You've changed the definition of the test, e.g. changing the size of
+//       the test buffers.
+//   (2) You've changed the implementation of memset, memcpy, etc. and they can
+//       do the same job in fewer cycles.
+//   (3) The test was not compiled with `-O2`. The hardcoded cycle count
+//       expectations assume `-O2`.
+//   (4) The compiler has gotten smarter.
+//
+// If you observe the cycle count is smaller the hardcoded expectation, that's
+// probably a good thing; please update the expectation!
+static const perf_test_t kPerfTests[] = {
+    {.label = "memcpy",
+     .setup_buf1 = &fill_buf_deterministic_values,
+     .setup_buf2 = &fill_buf_deterministic_values,
+     .func = &test_memcpy,
+     .expected_num_cycles = 10540544},
+    {.label = "memcpy_zeroes",
+     .setup_buf1 = &fill_buf_deterministic_values,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memcpy,
+     .expected_num_cycles = 10540544},
+    {.label = "memset",
+     .setup_buf1 = &fill_buf_zeroes,
+     .setup_buf2 = &fill_buf_deterministic_values,
+     .func = &test_memset,
+     .expected_num_cycles = 6539544},
+    {.label = "memset_zeroes",
+     .setup_buf1 = &fill_buf_zeroes,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memset,
+     .expected_num_cycles = 6539544},
+    {.label = "memcmp_pathological",
+     .setup_buf1 = &fill_buf_zeroes_then_one,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memcmp,
+     .expected_num_cycles = 10543547},
+    {.label = "memcmp_zeroes",
+     .setup_buf1 = &fill_buf_zeroes,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memcmp,
+     .expected_num_cycles = 10540547},
+    {.label = "memrcmp_pathological",
+     .setup_buf1 = &fill_buf_zeroes,
+     .setup_buf2 = &fill_buf_one_then_zeroes,
+     .func = &test_memrcmp,
+     .expected_num_cycles = 10045045},
+    {.label = "memrcmp_zeroes",
+     .setup_buf1 = &fill_buf_zeroes,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memrcmp,
+     .expected_num_cycles = 10042045},
+    {.label = "memchr_pathological",
+     .setup_buf1 = &fill_buf_deterministic_values,
+     .setup_buf2 = &fill_buf_zeroes,
+     .func = &test_memchr,
+     .expected_num_cycles = 2401080},
+    {.label = "memrchr_pathological",
+     .setup_buf1 = &fill_buf_deterministic_values,
+     .setup_buf2 = &fill_buf_deterministic_values,
+     .func = &test_memrchr,
+     .expected_num_cycles = 2160744},
+};
+
+bool test_main(void) {
+  uint8_t buf1[PERFTEST_BUF_LEN] = {0};
+  uint8_t buf2[PERFTEST_BUF_LEN] = {0};
```

Comment:
Any particular reason to put these on the stack and initialize them? (ie: if they were uninitialized globals, they wouldn't have to be constrained by some notion of stack size and they'd be guaranteed to be zero-initialized because they'd be in the `.bss` section).